examples/tv-app/linux/include/tv-channel/TvChannelManager.cpp

```diff
@@ -88,35 +96,47 @@ std::vector<EmberAfTvChannelInfo> TvChannelManager::proxyGetTvChannelList()
     return tvChannels;
 }
 
-static void sendResponse(const char * responseName, EmberAfTvChannelInfo channelInfo)
+void sendResponse(chip::app::Command * command, EmberAfTvChannelInfo channelInfo)
 {
-    emberAfFillExternalBuffer((ZCL_CLUSTER_SPECIFIC_COMMAND | ZCL_FRAME_CONTROL_SERVER_TO_CLIENT), ZCL_TV_CHANNEL_CLUSTER_ID,
-                              ZCL_CHANGE_CHANNEL_RESPONSE_COMMAND_ID, "uusss", channelInfo.majorNumber, channelInfo.minorNumber,
-                              channelInfo.affiliateCallSign, channelInfo.callSign, channelInfo.name);
-
-    EmberStatus status = emberAfSendResponse();
-    if (status != EMBER_SUCCESS)
+    CHIP_ERROR err                         = CHIP_NO_ERROR;
+    chip::app::CommandPathParams cmdParams = { emberAfCurrentEndpoint(), /* group id */ 0, ZCL_TV_CHANNEL_CLUSTER_ID,
+                                               ZCL_CHANGE_CHANNEL_RESPONSE_COMMAND_ID,
+                                               (chip::app::CommandPathFlags::kEndpointIdValid) };
+    chip::TLV::TLVWriter * writer          = nullptr;
+    SuccessOrExit(err = command->PrepareCommand(cmdParams));
+    VerifyOrExit((writer = command->GetCommandDataElementTLVWriter()) != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    SuccessOrExit(err = writer->Put(chip::TLV::ContextTag(0), channelInfo.majorNumber));
+    SuccessOrExit(err = writer->Put(chip::TLV::ContextTag(1), channelInfo.minorNumber));
+    // TODO: ByteSpan as TLVWriter param is not supported. Getting an error
```

Comment:
It should work fine.  TLVWriter has `CHIP_ERROR Put(uint64_t tag, ByteSpan data);` for putting an octet string...

Reply:
Ok, so the return value here should be:
- For index 0, Struct TvChannelInfo
- For index 1, Enum of potential error.

Per latest info I have struct as response is not supproted atm, so I've commented this out.

Reply:
> Per latest info I have struct as response is not supproted atm

Not supported in the sense that the decoder on the other side can't decode it?  The encoder can encode it fine.  It's just that with our current (bad) API you have to manually start a TLV struct on the writer, encode the struct members one by one, then close the struct.  

The decoder has no way to make sense of it, so will just error out.  But it will error out if it's not provided too, so if you care about the decoder something else needs to happen here....

Reply:
Discussed offline. 
Currently we're not able to return an array using the TLV writter:
`SuccessOrExit(err = writer->Put(chip::TLV::ContextTag(0), anExampleArray));`